capsules/src/uart.rs

```diff
@@ -55,45 +56,173 @@ pub struct App {
     read_len: usize,
 }
 
-pub static mut WRITE_BUF: [u8; 64] = [0; 64];
-pub static mut READ_BUF: [u8; 64] = [0; 64];
+pub static mut WRITE_BUF0: [u8; 64] = [0; 64];
+pub static mut READ_BUF0: [u8; 64] = [0; 64];
 
-pub struct Console<'a, U: UART> {
-    uart: &'a U,
-    apps: Grant<App>,
+pub static mut WRITE_BUF1: [u8; 64] = [0; 64];
+pub static mut READ_BUF1: [u8; 64] = [0; 64];
+
+pub struct UartDriver<'a, U: 'static + hil::uart::UART> {
+    uarts: &'a mut [&'a mut Uart<'a, U>],
+    apps: [Grant<App>; 2],
+}
+
+impl<'a, U: 'static + hil::uart::UART> UartDriver<'a, U> {
+    pub fn new(
+        uarts: &'a mut [&'static mut Uart<'a, U>],
```

Comment:
This _could_ be a `AsRef<[&'static mut Uart[<'a, U>]>`, which would allow the array itself to be owned by the UartDriver, so it doesn't have to be allocated somewhere else (since arrays up to size 16 or something implement `AsRef`). Not a particularly important difference, just maybe aesthetically a bit different and probably slightly easier to initialize.